hypershift-operator/controllers/nodepool/nodepool_controller.go

```diff
@@ -1264,18 +1264,23 @@ func (r *NodePoolReconciler) deleteKubeVirtCache(ctx context.Context, nodePool *
 	if nodePool.Status.Platform != nil {
 		if nodePool.Status.Platform.KubeVirt != nil {
 			if cacheName := nodePool.Status.Platform.KubeVirt.CacheName; len(cacheName) > 0 {
-				ns := controlPlaneNamespace
+				uid := string(nodePool.GetUID())
+				cl, err := r.KubevirtInfraClients.DiscoverKubevirtClusterClient(ctx, r.Client, uid, nodePool.Status.Platform.KubeVirt.Credentials, controlPlaneNamespace, nodePool.GetNamespace())
+				if err != nil {
+					return fmt.Errorf("failed to get KubeVirt external infra-cluster:  %w", err)
+				}
+				if cl != nil {
+					r.KubevirtInfraClients.Delete(uid)
```

Comment:
I believe there is a memory leak with `r.KubevirtInfraClients.Delete(uid)`. We're creating the client unconditionally, but only deleting the client when the cache is used.

Reply:
done